Frameworks/Foundation/NSURLProtocol_WinHTTP.mm

```diff
@@ -288,7 +288,8 @@ - (void)startLoading {
             // TODO #2721: If this properly supports streaming bodies instead of flattening the data up front,
             // more accurate callbacks can be done
             NSInteger contentSize = self.request.HTTPBody ? self.request.HTTPBody.length : _flattenedBodyStream.size();
-            if (contentSize > 0) {
+            if (([self.client respondsToSelector:@selector(URLProtocol:didWriteData:totalBytesWritten:totalBytesExpectedToWrite:)]) &&
```

Comment:
In looking for this issue, I realized that no header declares the protocol method `URLProtocol:didWriteData:...`; could we add one in NSURLProtocolInternal? Ideally, NSURLConnection would conform to this private protocol, and the compiler would warn us if it didn't have the right methods on it. NSURLSessionTask _should_ eventually do this, but it doesn't.